ui/src/main/java/edu/wpi/grip/ui/Main.java

```diff
@@ -76,35 +77,28 @@ public void start(Stage stage) throws Exception {
       injector.injectMembers(this);
 
       parameters.remove("--headless");
+      headless = true;
     } else {
       // Otherwise, run with both the core and UI modules, and show the JavaFX stage
       injector = Guice.createInjector(Modules.override(new GripCoreModule(), new GripFileModule(),
           new GripSourcesHardwareModule()).with(new GripNetworkModule(), new GripUiModule()));
-
       injector.injectMembers(this);
+      notifyPreloader(new Preloader.ProgressNotification(0.15));
```

Comment:
I can guarantee you that creating the dependency injector probably takes the most amount of time.